client/state/selectors/is-jetpack-onboarding-step-completed.js

```diff
@@ -0,0 +1,48 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import createSelector from 'lib/create-selector';
+import { getJetpackOnboardingSettings } from 'state/selectors';
+import { JETPACK_ONBOARDING_STEPS as STEPS } from 'jetpack-onboarding/constants';
+
+export const isJetpackOnboardingStepCompleted = createSelector(
+	( state, siteId, stepName ) => {
+		const settings = getJetpackOnboardingSettings( state, siteId );
+
+		if ( ! settings ) {
+			return null;
```

Comment:
Oh the old dilemma: Should an `is...` selector return a non-bool when it doesn't have data available? I think I'd argue it shouldn't. Most often, they will be used by the UI to represent some sort of state. If we haven't obtained the required information yet, we should indicate a 'loading' state, but based upon information from a dedicated `isRequesting...` selector (or the new data-layer counterpart, see #20038), rather then sort of extending the return type of the `is...` selector to also allow `null`, and checking for `=== false` or `=== null` which IMO is plain ugly.

Reply:
Makes sense 👍 

Updated in https://github.com/Automattic/wp-calypso/pull/21438/commits/811bbc7574ce913374872a8e2d71d352112e9b54.

Reply:
For completeness' sake, pasting my extended argument from Slack here:

> I think the correct approach to think about it is: in which cases are we using it? Is it a magic value that consistently indicates something that can be reliably indicated that way, or not?
> Another angle: Is this only possible with boolean (`is...`) selectors, or does it work with non-boolean ones (typically `get...`), too?
> For boolean ones, it’s arguably a tertiary value. That’s at least a somewhat familiar pattern.
> Let’s look at all the cases where we return `null` from this selector, and what they mean:
> https://github.com/Automattic/wp-calypso/pull/21438/files#diff-cf16f5fa16dee3151e0eebdf918d13d8R21
> https://github.com/Automattic/wp-calypso/pull/21438/files#diff-cf16f5fa16dee3151e0eebdf918d13d8R43
> And consequently, what we want to display in the UI in each case.
> So on R21, we most likely haven’t fetched any data yet for the given site; unless we’ve erroneously specified a wrong `siteId` altogether.
> So in the ‘sane’ case, we haven’t fetched yet, so we’d want to show a loading spinner.
> But what about R43? That only happens if we’ve specified a non-existent `stepName`, correct? 
> So that’s a non-welldefined (non-sane) case by definition.
> So showing a spinner in this case wouldn’t really make any sense.
> And that kind of concludes my argument: The `null` return value isn’t any more expressive than simply using `false` here; determining whether we’re still loading cannot be sufficiently accurately expressed here (unless we go to even greater lengths and use a fourth possible return value such as `undefined` in one of the cases described here, but that would just make it more messy IMO).